### PR TITLE
[Server] WMS GetContext refactoring

### PR DESCRIPTION
**Pull request by rldhont**
Removing QgsWMSProjectParser from GetContext
https://github.com/qgis/qgis3.0_api/issues/57

## Checklist
- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/server/services/wms/qgswmsgetcontext.cpp [hunk 1]
```diff
@@ -64,11 +91,376 @@ namespace QgsWms
     owsContextElem.setAttribute( QStringLiteral( "version" ), QStringLiteral( "0.3.1" ) );
     doc.appendChild( owsContextElem );
 
-    QString hrefString = serviceUrl( request, project ).toString( QUrl::FullyDecoded );
-    configParser->owsGeneralAndResourceList( owsContextElem, doc, hrefString );
+    //QString hrefString = serviceUrl( request, project ).toString( QUrl::FullyDecoded );
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetcontext.cpp]: Any reason not to just kill these commented out blocks?

File: src/server/services/wms/qgswmsgetcontext.cpp [hunk 2]
```diff
@@ -64,11 +91,376 @@ namespace QgsWms
...
+      }
+
+      // OWSContext General element is complete
+      parentElement.appendChild( generalElem );
+
+      // OWSContext ResourceList element
+      QDomElement resourceListElem = doc.createElement( QStringLiteral( "ResourceList" ) );
+      const QgsLayerTree *projectLayerTreeRoot = project->layerTreeRoot();
+      QgsRectangle combinedBBox;
+      appendOwsLayersFromTreeGroup( doc, resourceListElem, serverIface, project, request, projectLayerTreeRoot, combinedBBox, QLatin1String( "" ) );
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetcontext.cpp]: Use QString() instead of QLatin1String("")

**Review by nyalldawson — APPROVED** [review]
Code looks good to me!

File: src/server/services/wms/qgswmsgetcontext.cpp (hunk 2 above)
Comment (rldhont) [inline src/server/services/wms/qgswmsgetcontext.cpp]: I'll copy it from old code, I'll fix it, thanks.

File: src/server/services/wms/qgswmsgetcontext.cpp (hunk 1 above)
Comment (rldhont) [inline src/server/services/wms/qgswmsgetcontext.cpp]: It's just an oversight. Thanks, I'll fix it

**Pull request merged**